praktikum/poglavlja/12_dinamicka_alokacija/primeri/realloc.c

```diff
@@ -0,0 +1,63 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+int main()
+{
+    int *a, n;
+
+    do
+    {
+        printf("Unesite duzinu niza: ");
+        scanf("%d", &n);
+
+    }while(n <= 0);
+
+    a = (int *) malloc(n * sizeof(int));
+
+    if (a == NULL)
+    {
+        printf("Memorija se ne moze zauzeti.\n");
+        exit(EXIT_FAILURE);
+    }
+
+    for(int i = 0; i < n; i++)
+    {
+        printf("a[%d] = ", i);
+        scanf("%d", &a[i]);
+    }
+
+    int m;
+
+    do
+    {
+        printf("Unesite novu duzinu niza: ");
+        scanf("%d", &m);
+
+    }while(m <= 0);
+
+    a = (int *) realloc(a, m * sizeof(int));
```

Comment:
Da li si ovo dobro istestirala? Ja sam uglavnom imao problema kad sam direktno gazio vrednost pokazivačke promenljive čiju prethodnu vrednost sam ujedno prosledio ``realloc``-u. Uglavnom mi je bila potrebna neka tmp vrednost u koju sam smestio rezultat ``realloc``-a, pa onda dodelim u nekoj sledećoj liniji nazad na ``a``.

Reply:
Da, i kod mene radi sve.

Reply:
Više na [ovom linku](https://stackoverflow.com/questions/72332551/is-it-mandatory-to-check-if-realloc-worked). Ako ``realloc`` nije uspešan, izgubiće se pokazivač ka starom memorijskom bloku. Jedna od varijanti je da se uvede privremena pokazivačka promenljiva, koja će se prethodno proveriti, pre nego što se pregazi vrednost stare, a drugu vidiš na linku koji sam ti poslao, sa ``assert`` naredbom, koja će prekinuti program u potpunosti. Više sam fan ovog prvog pristupa.